main/src/ca/uwaterloo/flix/language/ast/Purity.scala

```diff
@@ -32,4 +32,15 @@ object Purity {
    * Represents an impure expression (i.e. an expression that could potentially have side-effects).
    */
   case object Impure extends Purity
+
+  /**
+   * Infix operator `combine` merges purities `self` and `that`
+   * A merged purity is only pure if both `self` and `that` are pure, otherwise it is always impure.
+   */
+  implicit class compare(self: Purity) extends Purity {
```

Comment:
We never use implicit classes. They are really a misfeature.

Where is this used and can it not be a normal function?

Reply:
This was a result of my suggestion. I thought an infix would be a clean syntax for combining multiple purities. Though I meant to add it as a method in the Purity trait :)

Reply:
We can add it to the purity trait, but I prefer just a good ol' regular function somewhere.